src/main/java/com/azure/cosmos/examples/throughputcontrol/async/ThroughputControlQuickstartAsync.java

```diff
@@ -0,0 +1,322 @@
+                        .groupName("globalControlGroup")
+                        // use below to set a target throughput threshold as % of provisioned throughput instead of absolute value
+                        //.targetThroughputThreshold(0.25)
+                        .targetThroughput(THROUGHPUT_CONTROL_RUS)
+                        .build();
+        GlobalThroughputControlConfig globalControlConfig1 =
+                this.client1.createGlobalThroughputControlConfigBuilder(database1.getId(), throughputControlContainerName)
+                        .setControlItemRenewInterval(Duration.ofSeconds(5))
+                        .setControlItemExpireInterval(Duration.ofSeconds(20))
+                        .build();
+        GlobalThroughputControlConfig globalControlConfig2 =
+                this.client2.createGlobalThroughputControlConfigBuilder(database1.getId(), throughputControlContainerName)
+                        .setControlItemRenewInterval(Duration.ofSeconds(5))
+                        .setControlItemExpireInterval(Duration.ofSeconds(20))
+                        .build();
+        ThroughputControlTestContainerObject1.enableGlobalThroughputControlGroup(groupConfig, globalControlConfig1);
+        ThroughputControlTestContainerObject2.enableGlobalThroughputControlGroup(groupConfig, globalControlConfig2);
+        try {
+            createManyItemsWithTwoClients(NUMBER_OF_DOCS, "GLOBAL THROUGHPUT CONTROL TEST", options, Arrays.asList(ThroughputControlTestContainerObject1, ThroughputControlTestContainerObject2));
+        } catch (Exception e) {
+            logger.info("Exception in globalThroughputControlTest: " + e);
+        }
+    }
+
+    /**
+     * Priority based throttling test - one client will have priority of LOW, the other HIGH
+     */
+    private void priorityBasedThrottling() {
+        database1 = client1.getDatabase(databaseName);
+        database2 = client2.getDatabase(databaseName);
+        priorityBasedThrottlingContainerObject1 = database1.getContainer(priorityBasedThrottlingContainerName);
+        priorityBasedThrottlingContainerObject2 = database2.getContainer(priorityBasedThrottlingContainerName);
+        CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        options.setThroughputControlGroupName("priorityBasedThrottling");
+        ThroughputControlGroupConfig groupConfig1 =
+                new ThroughputControlGroupConfigBuilder()
+                        .groupName("priorityBasedThrottling")
+                        .priorityLevel(PriorityLevel.HIGH)
+                        .build();
+        ThroughputControlGroupConfig groupConfig2 =
+                new ThroughputControlGroupConfigBuilder()
+                        .groupName("priorityBasedThrottling")
+                        .priorityLevel(PriorityLevel.LOW)
+                        .build();
+        ThroughputControlTestContainerObject1.enableLocalThroughputControlGroup(groupConfig1);
```

Comment:
I only avoided this because right now I believe the priority will only apply to backend rate limiting, and not SDK rate limiting controlled by throughput control, which I feel could make things a bit misleading (customer might not be sure where the priority is being applied). At a later point in time, when throughput control is provided by backend (as part of extending priority based throttling), then I think this will make more sense?